### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by sneeu-leeu**
You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
Login page and registration page:
Should be built with Devise.
Food list:
Should display a list of food as in the wireframe.
Should lead to the form that allows users to add new food.
Recipies list:
Should display a list of recipes created by the logged-in user as in the wireframe.
Should lead to recipe details.
If the user is the owner of the recipe, should allow user to delete it.
Public recipe list:
Should display a list of all public recipes ordered by newest as in the wireframe.
Should lead to recipe details.
Recipe details:
If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
If the user is the owner of the recipe, should lead to the form that allows user to add new food.
General shopping list view (do not implement if in group of 3):
Should show the list of food that is missing for all recipes of the logged-in user.
Should count the total food items and total price of the missing food.
Make sure there are no N+1 queries happening.
Create navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  devise_for :users
+  root 'foods#index'
+  get 'users/index'
+  get 'users/sign_out'
+  resources :users, only: %i[index] 
+  resources :foods, only: %i[index new create destroy]
+  resources :recipes, only: [:index, :show, :new, :create, :destroy] do
+    resources :recipe_foods, only: [:new, :create, :destroy, :update, :edit]
```
Comment (uwadonat) [inline config/routes.rb:9]: - [x] User(owner) should be able to delete a food item from the lists. Currently, it is not working.

Screenshot from 2022-03-19 19-29-22

File: app/views/recipes/public.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <div class="recipe">
+    <div class="recipe-name">
+      <div>
+      <%= link_to "#{recipe.name}", recipe_path(recipe), class: "text-reset text-decoration-none fs-3 text-capitalize" %>
+      </div>
+    </div>
+    <div class="description">
+      <p><%= recipe.description %></p>
+    </div>
+  </div>
```
Comment (uwadonat) [inline app/views/recipes/public.html.erb]: - [x] Let's add the recipe's owner to match the design
Screenshot from 2022-03-19 19-50-36

Comment (uwadonat) [reply app/views/recipes/public.html.erb]: - [x] Kindly add some styling. Your project will look more professional.

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **@Team**, :hand: 

Good job so far! :mechanical_arm: 
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight! :+1: 
- Your commit messages are very descriptive ✔️
- No linter errors ✔️
- You tested your project ✔️

## Status: Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/foods_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @food = Food.new(food_params)
+    if @food.save
+      flash[:notice] = 'Food created successfully.'
+      redirect_to foods_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
```
Comment (SajjadAhmad14) [inline app/controllers/foods_controller.rb:21]: The user gets an error when trying to delete a food by another user.
delete

File: app/controllers/recipes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    if @recipe.save
+      flash[:notice] = 'Recipe successfully created'
+      redirect_to @recipe
+    else
+      flash[:notice] = 'Something went wrong'
+      render 'new'
+    end
+  end
+
+  def destroy
```
Comment (SajjadAhmad14) [inline app/controllers/recipes_controller.rb:27]: [ OPTIONAL ] Users should only be allowed to delete their own recipes, not other users' recipes.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi team, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**